os/arch/arm/src/armv7-m/up_hardfault.c

```diff
@@ -137,14 +138,13 @@ int up_hardfault(int irq, FAR void *context, FAR void *arg)
 	 * FLASH region or in the user FLASH region.
 	 */
 
-	if (((uintptr_t)pc >= (uintptr_t)&_stext && (uintptr_t)pc < (uintptr_t)&_etext) ||
-			(sched_self()->uspace &&
+	if ((is_kernel_text_space((void *)pc)) || (sched_self()->uspace &&
```

Comment:
`is_kernel_text_space` checks not only FLASH but also RAM space. Please update the comment also.